packages/inter-protocol/src/auction/scheduler.js

```diff
@@ -205,11 +213,12 @@ export const makeScheduler = async (
     publishSchedule();
   };
 
+  // XXX liveSchedule and nextSchedule are in scope
   // schedule the wakeups for the steps of this round
-  const scheduleSteps = () => {
-    if (!liveSchedule) throw Fail`liveSchedule not defined`;
+  const scheduleSteps = schedule => {
+    schedule || Fail`liveSchedule not defined`;
```

Comment:
_fix_:
```suggestion
    schedule || Fail`schedule not defined`;
```

what is the overall error handling strategy in this code? is it okay to call `Fail` at any point or are there strict rules on when to call `Fail` that could be noted in a comment somewhere?

i'm considering the safety/likelihood of this `Fail` ever firing. Currently `startAuction` is the only caller of `scheduleSteps` and it ensures that the value it passes for `schedule` is not undefined. if via refactoring this were to change and the `Fail` were to file what would the impact be?

```
    if (!nextSchedule) {
      // nothing new to schedule
      return;
    }
    // activate the nextSchedule as the live one
    // (read here and in function calls below)
    liveSchedule = nextSchedule;
    nextSchedule = safelyComputeRoundTiming(
      params,
      TimeMath.addAbsRel(liveSchedule.endTime, relativeTime(1n)),
    );

    scheduleSteps(liveSchedule);
```

Reply:
> what is the overall error handling strategy in this code? is it okay to call Fail at any point or are there strict rules on when to call Fail that could be noted in a comment somewhere?

It's a good question. I don't have a complete answer off the top of my head. I'm willing to spend time writing this up if it's a priority.

> if via refactoring this were to change and the Fail were to file what would the impact be?

`scheduleSteps` is responsible for setting up the timer that will wake the auction for each descending step. If there's no schedule when it's called, that waker won't be scheduled, and that round of the auction will pass without any bids being processed. If the caller is in a delicate place, the exception could propagate. But If it were called without no schedule, it wouldn't be able to do anything anyway. I can see an argument that it would be more robust to *return* rather than to *Fail*.

Reply:
> > what is the overall error handling strategy in this code? is it okay to call Fail at any point or are there strict rules on when to call Fail that could be noted in a comment somewhere?
> 
> It's a good question. I don't have a complete answer off the top of my head. I'm willing to spend time writing this up if it's a priority.

I think it's paramount for this change that an understood and consistent error handling strategy is employed. Perhaps @turadg has codified that ruleset?

Reply:
It should never be undefined at this call time and the reason for the param was that the static type analysis verified it. But to be more defensive I have it just log and return now.